src/cocotb/task.py

```diff
@@ -105,11 +111,40 @@ def __init__(
         self._done_callbacks: List[Callable[[Task[ResultType]], None]] = []
         self._cancelled_msg: Union[str, None] = None
         self._must_cancel: bool = False
+        self._locals = SimpleNamespace()
 
         self._task_id = self._id_count
         type(self)._id_count += 1
         self._name = f"Task {self._task_id}" if name is None else name
 
+    @property
+    def locals(self) -> SimpleNamespace:
+        '''Task-local variables.
```

Comment:
nit: should docstrings all use `"""` ?

Reply:
There's a `"""` internally. I think using `'''` is better than doing `\"\"\"`.